### PR TITLE
Admin topic rework. Moving to switch setup. ELSE IF CHAIN BEGONE!

### PR DESCRIPTION
**Pull request by Jalleo**
Done up to line 575 so far I will do more tonight/tomorrow. Probably up to 1500 odd.

Also adds feedback to end round button when you try to use it before the round has started.

:cl: Jalleo
admin: Admin topic might actually be "faster" with certain aspects.
/:cl:

### CODE REVIEW

**Comment by Jalleo** [comment]
By the way it technically can be testmerged with how many lines I have done so far already.

**Comment by k-k-karma** [comment]
stop removing our advanced 1500 line artifical banning intelligence

**Comment by Jalleo** [comment]
Somehow this is working on the version of byond that I have I am trying to work out the cause and if its like this for all versions.

**Comment by Jalleo** [comment]
I have realised already with the ahelp section. The very first one. I am going to update the PR now with comments explaining what I did.

File: code/modules/admin/topic.dm [hunk 1]
```diff
@@ -13,566 +13,568 @@
...
 	..()
-
 	if(usr.client != src.owner || !check_rights(0))
 		message_admins("[usr.key] has attempted to override the admin panel!")
 		log_admin("[key_name(usr)] tried to use the admin panel without authorization.")
 		return
 
 	if(!CheckAdminHref(href, href_list))
 		return
+	switch(href_list["admin_action"])
```
Comment (Jalleo) [inline code/modules/admin/topic.dm]: I have set it up so it looks for admin_action in the href_list.
admin_action will have the name of whichever admin topic it should be going to saved in it.
Example
`admin_action=stickyban; stickyban=add`

**Comment by Jalleo** [comment]
I cant confirm if the banning panel is working under my changes so far as I cannot work out how to setup the database to test it locally

**Comment by Cyberboss** [comment]
https://tgstation13.download/database/mariadb-v10.2.10-x86-trimmed-preprimed%28v4.4%29.zip

@MrStonedOne this needs updating

**Comment by optimumtact** [comment]
at some point, we need to bite the bullet and datumize each individual action in the admin href.

**Comment by Jalleo** [comment]
Okay I got that form bit resolved and will push the banning panel working. After that I will see about datums and how i could do that instead of just switching it all.

**Comment by subject217** [comment]
Merge conflicts? Stale?

**Comment by Jordie0608** [comment]
Rip this I guess.

**Comment by Jalleo** [comment]
Stale I kinda forgot about it honestly. I will update my local and see about working on it once I worked out what to do. Then open a new PR smaller scale but working compared to the incomplete code here.

**Pull request closed without merging**